cosima_cookbook/database.py

```diff
@@ -626,14 +653,9 @@ def prune_experiment(experiment, session, delete=True):
         print("No such experiment: {}".format(experiment))
         return
 
-    for f in expt.ncfiles:
-        # check whether file exists
-        if not f.ncfile_path.exists() or not f.present:
+    files = {f for f in expt.ncfiles if not f.present or not f.ncfile_path.exists()}
```

Comment:
With the proposed idea above, `prune_files` should be module private (i.e. `_prune_files`), and this gets a similar structure. I can't see a clean way of incorporating the `.exists()` check, so I guess this would just be another `.in_()` expression? It's a bit unsatisfying to have to bring the whole query across and then translate it back!

Reply:
New logic removes the need for this.